main.go

```diff
@@ -314,95 +305,73 @@ func main() {
 				log.Error(err, "unable to create webhook", "webhook", "RunnerReplicaSet")
 				os.Exit(1)
 			}
+			injector := &actionssummerwindnet.PodRunnerTokenInjector{
+				Client:       mgr.GetClient(),
+				GitHubClient: multiClient,
+				Log:          ctrl.Log.WithName("webhook").WithName("PodRunnerTokenInjector"),
+			}
+			if err = injector.SetupWithManager(mgr); err != nil {
+				log.Error(err, "unable to create webhook server", "webhook", "PodRunnerTokenInjector")
+				os.Exit(1)
+			}
 		}
 	}
 
-	// We use this environment avariable to turn on the ScaleSet related controllers.
-	// Otherwise ARC's legacy chart is unable to deploy a working ARC controller-manager pod,
-	// due to that the chart does not contain new actions.* CRDs while ARC requires those CRDs.
-	//
-	// We might have used a more explicitly named environment variable for this,
-	// e.g. "CONTROLLER_MANAGER_ENABLE_SCALE_SET" to explicitly enable the new controllers,
-	// or "CONTROLLER_MANAGER_DISABLE_SCALE_SET" to explicitly disable the new controllers.
-	// However, doing so would affect either private ARC testers or current ARC users
-	// who run ARC without those variabls.
-	mgrPodName := os.Getenv("CONTROLLER_MANAGER_POD_NAME")
-	if mgrPodName != "" {
-		mgrPodNamespace := os.Getenv("CONTROLLER_MANAGER_POD_NAMESPACE")
-		var mgrPod corev1.Pod
-		err = mgr.GetAPIReader().Get(context.Background(), types.NamespacedName{Namespace: mgrPodNamespace, Name: mgrPodName}, &mgrPod)
-		if err != nil {
-			log.Error(err, fmt.Sprintf("unable to obtain manager pod: %s (%s)", mgrPodName, mgrPodNamespace))
-			os.Exit(1)
-		}
-
-		var mgrContainer *corev1.Container
-		for _, container := range mgrPod.Spec.Containers {
-			if container.Name == "manager" {
-				mgrContainer = &container
-				break
-			}
-		}
+	managerImage := os.Getenv("CONTROLLER_MANAGER_CONTAINER_IMAGE")
```

Comment:
should we wrap all of these into `if autoScalingRunnerSetOnly {}`? otherwise, the old `actions-runner-controller` helm chart installation won't start the controller correctly.